functions/helperFunctions.sh

```diff
@@ -444,4 +444,21 @@ function createDesktopShortcut(){
 	chmod +x "$Shortcutlocation"
 
 	echo "$Shortcutlocation created"
+}
+
+# Migrate dolphin save states files from 0.17.6 location to new location
+function migrateDolphinStates() {
+	local emuName=$1
+	local flatpakName=$2
+	local oldStatesPath="$HOME/.var/app/$flatpakName/data/dolphin-emu/states"
+	local newStatesPath="$HOME/.var/app/$flatpakName/data/dolphin-emu/StateSaves"
+	local linkedTarget=$(readlink -f "$savesPath/$emuName/states")
+
+	if [[ -e "$oldStatesPath" && "$oldStatesPath" == "$linkedTarget" ]]; then
+		echo "Migrating $emuName states from $oldStatesPath to $newStatesPath"
+		cp -r "$oldStatesPath/*" "$newStatesPath"
+		rm -rf "$oldStatesPath"
+		unlink "$savesPath/$emuName/states"
+		moveSaveFolder dolphin states "$HOME/.var/app/$flatpakName/data/dolphin-emu/StateSaves"
```

Comment:
this will not work for primehack, because it will try to link it to the Emulation/dolphin/states folder